packages/perseus/src/widget-ai-utils/interactive-graph/prompt-utils.ts

```diff
@@ -0,0 +1,295 @@
+            } as PolygonGraphOptions;
+            break;
+        case "quadratic":
+            graphOptions = {
+                type: props.graph.type,
+                startCoords: props.graph.startCoords,
+            } as QuadraticGraphOptions;
+            break;
+        case "ray":
+            graphOptions = {
+                type: props.graph.type,
+                startCoords: props.graph.startCoords,
+            } as RayGraphOptions;
+            break;
+        case "segment":
+            graphOptions = {
+                type: props.graph.type,
+                numSegments: props.graph.numSegments,
+                startCoords: props.graph.startCoords,
+            } as SegmentGraphOptions;
+            break;
+        case "sinusoid":
+            graphOptions = {
+                type: props.graph.type,
+                startCoords: props.graph.startCoords,
+            } as SinusoidGraphOptions;
+            break;
+        default:
+            return getUnsupportedPromptJSON(
+                "interactive-graph-unsupported",
+                `The graph type ${props.graph.type} is not supported.`,
+            );
+    }
+
+    switch (userInput.type) {
+        case "angle":
+            input = {
+                coords: userInput.coords,
+                angleOffsetDegrees: userInput.angleOffsetDeg,
+            } as AngleUserInput;
+            break;
+        case "circle":
+            input = {
+                center: userInput.center,
+                radius: userInput.radius,
+            } as CircleUserInput;
+            break;
+        case "linear":
+            input = {
+                coords: userInput.coords,
+            } as LinearUserInput;
+            break;
+        case "linear-system":
+            input = {
+                coords: userInput.coords,
+            } as LinearSystemInput;
+            break;
+        case "point":
+            input = {
+                coords: userInput.coords,
+            } as PointUserInput;
+            break;
+        case "polygon":
+            input = {
+                coords: userInput.coords,
+            } as PolygonUserInput;
+            break;
+        case "quadratic":
+            input = {
+                coords: userInput.coords,
+            } as QuadraticUserInput;
+            break;
+        case "ray":
+            input = {
+                coords: userInput.coords,
+            } as RayUserInput;
+            break;
+        case "segment":
+            input = {
+                coords: userInput.coords,
+            } as SegmentUserInput;
+            break;
+        case "sinusoid":
+            input = {
+                coords: userInput.coords,
+            } as SinusoidUserInput;
+            break;
+        default:
+            return getUnsupportedPromptJSON(
+                "interactive-graph-unsupported",
+                `The graph type ${props.graph.type} is not supported.`,
+            );
```

Comment:
I think we'd be better served leveraging TypeScript here and avoiding a message that I'm skeptical will be used. :) 

What if we just `throw new UnreachableCaseError(userInput.type);` here? You can see how it's used [here](https://github.com/Khan/perseus/blob/22d1c0229e52fc2dc6bc9f38a3ca400c6e4a3884/packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx#L645). 

Note that we might need to assign the type to a const before the switch due to how TypeScript narrowing works with object properties (or rather, how it doesn't work).